controllers/cluster_controller.go

```diff
@@ -262,9 +270,135 @@ func (r *ClusterReconciler) postClusterProviderReconcile(ctx context.Context, lo
 		}
 	}
 
+	if cluster.IsPackagesEnabled() {
+		if err := r.postReconcilePackagesForWorkloadCluster(ctx, log, cluster); err != nil {
```

Comment:
does this need to happen for both management and workload cluster?
if it's only for workload then I would do
```suggestion
	if cluster.IsPackagesEnabled() && cluster.IsManaged() {
		if err := r.postReconcilePackagesForWorkloadCluster(ctx, log, cluster); err != nil {
```

Reply:
It needs only run for workload clusters. Packages installation via FCL isn't supported for management clusters yet, as FCL isn't supported for management clusters yet. :)

IsPackagesEnabled is simply:

```return !c.IsSelfManaged() && (c.Spec.Packages == nil || !c.Spec.Packages.Disable)```

I felt that extracting the proposition to its own method read better.